packages/cli/__tests__/zosjobs/__system__/cancel/cli.zos-jobs.cancel.job.system.test.ts

```diff
@@ -49,15 +49,32 @@ describe("zos-jobs cancel job command", () => {
             expect(response.stderr.toString()).toContain("Obtaining job info for a single job id JOB00000 on");
             expect(response.stderr.toString()).toContain("failed: Job not found");
         });
+
+        it("should surface an error from z/OSMF if the jobid was already cancelled", () => {
+            runCliScript(__dirname + "/__scripts__/job/submit_job.sh", TEST_ENVIRONMENT, [LOCAL_JCL_FILE]);
+            const response = runCliScript(__dirname + "/__scripts__/job/cancel_job_v2_bad.sh", TEST_ENVIRONMENT, [LOCAL_JCL_FILE]);
+            expect(response.status).toBe(1);
+            expect(response.stderr.toString()).toContain("Failed to cancel job");
+            expect(response.stderr.toString()).toContain("Job not cancellable or purgeable");
+            expect(response.stderr.toString()).toContain("returned on second cancel");
+            expect(response.stdout.toString()).toContain("Successfully cancelled job");
```

Comment:
This is the test that is probably the most confusing, so that is understandable. The shell script in this instance runs three Zowe commands in this order:

1. Submit job and parse out the job number.
2. Using the job number from 1, cancel the job.
3. Repeat 2.

The shell script is running the cancel command twice. The first one should succeed, the second one should fail. The check for "returned on second cancel" in stderr verifies that the second cancel command is the one that failed. This line verifies (by elimination) that the first one succeeded.

Reply:
This makes a lot of sense now, thanks for the explanation 🙂